components/sections/NavigationBar/index.tsx

```diff
@@ -0,0 +1,95 @@
+import Hamburger from '@heroicons/react/24/solid/Bars3Icon';
+import XMark from '@heroicons/react/24/solid/XMarkIcon';
+import * as Label from '@radix-ui/react-label';
+import classNames from 'classnames';
+import Image from 'next/image';
+import { type FC, type ComponentProps, useState } from 'react';
+
+import LanguageDropdown from '@/components/Common/LanguageDropDown';
+import ThemeToggle from '@/components/Common/ThemeToggle';
+import Github from '@/components/Icons/Github';
+import NavItem from '@/components/sections/NavItem';
+import { BASE_PATH } from '@/next.constants.mjs';
+
+import style from './index.module.css';
+type NavItem = { text: string; href: string };
+
+type NavbarProps = {
+  navItems: NavItem[];
+  currentNavItem: string;
+  languages: ComponentProps<typeof LanguageDropdown>;
+  onThemeTogglerClick: () => void;
+};
+
+const navInteractionIcons = {
+  show: <Hamburger className="h-6 w-6" />,
+  close: <XMark className="h-6 w-6" />,
+};
+
+const NavigationBar: FC<NavbarProps> = ({
+  navItems,
+  currentNavItem,
+  languages,
+  onThemeTogglerClick,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
```

Comment:
Can we avoid using states here? Like I wonder if we can just with pure CSS keep track if the label was clicked or not. I assume maybe we can use the `:active` state? IDK

The reason why I am "cringing" over the "useState" statement, is that this component would need to become a Client-side component (aka you need to add `"use client";` at the beginning of this component, once you rebase this branch.